lib/matplotlib/figure.py

```diff
@@ -1528,635 +965,1865 @@ def subplots(self, nrows=1, ncols=1, sharex=False, sharey=False,
 
-        gridspec_kw : dict, optional
-            Dictionary with keywords passed to the `.GridSpec` constructor used
-            to create the grid the subplots are placed on.
+        The elements to be added to the legend are automatically determined,
+        when you do not pass in any extra arguments.
 
-        empty_sentinel : object, optional
-            Entry in the layout to mean "leave this space empty".  Defaults
-            to ``'.'``. Note, if *layout* is a string, it is processed via
-            `inspect.cleandoc` to remove leading white space, which may
-            interfere with using white-space as the empty sentinel.
+        In this case, the labels are taken from the artist. You can specify
+        them either at artist creation or by calling the
+        :meth:`~.Artist.set_label` method on the artist::
 
-        Returns
-        -------
-        dict[label, Axes]
-           A dictionary mapping the labels to the Axes objects.
+            ax.plot([1, 2, 3], label='Inline label')
+            fig.legend()
 
-        """
-        subplot_kw = subplot_kw or {}
-        gridspec_kw = gridspec_kw or {}
-        # special-case string input
-        if isinstance(layout, str):
-            layout = self._normalize_grid_string(layout)
+        or::
 
-        def _make_array(inp):
-            """
-            Convert input into 2D array
+            line, = ax.plot([1, 2, 3])
+            line.set_label('Label via method')
+            fig.legend()
 
-            We need to have this internal function rather than
-            ``np.asarray(..., dtype=object)`` so that a list of lists
-            of lists does not get converted to an array of dimension >
-            2
+        Specific lines can be excluded from the automatic legend element
+        selection by defining a label starting with an underscore.
+        This is default for all artists, so calling `.Figure.legend` without
+        any arguments and without setting the labels manually will result in
+        no legend being drawn.
 
-            Returns
-            -------
-            2D object array
 
-            """
-            r0, *rest = inp
-            for j, r in enumerate(rest, start=1):
-                if isinstance(r, str):
-                    raise ValueError('List layout specification must be 2D')
-                if len(r0) != len(r):
-                    raise ValueError(
-                        "All of the rows must be the same length, however "
-                        f"the first row ({r0!r}) has length {len(r0)} "
-                        f"and row {j} ({r!r}) has length {len(r)}."
-                    )
-            out = np.zeros((len(inp), len(r0)), dtype=object)
-            for j, r in enumerate(inp):
-                for k, v in enumerate(r):
-                    out[j, k] = v
-            return out
+        **2. Labeling existing plot elements**
+        To make a legend for all artists on all Axes, call this function with
```

Comment:
There was a blank line here.
```suggestion
        **2. Labeling existing plot elements**

        To make a legend for all artists on all Axes, call this function with
```